docpages/advanced_reference/coroutines.md

```diff
@@ -19,7 +19,7 @@ int main() {
 
     /* Message handler to look for a command called !file */
     /* Make note of passing the event by value, this is important (explained below) */
-    bot.on_message_create.co_attach([](dpp::message_create_t event) -> dpp::task<void> {
+    bot.on_message_create.co_attach([](dpp::message_create_t event) -> dpp::job {

```

Comment:
Not sure i'm right, but should this not be `const dpp::message_create_t& event` or does that not work with coroutines?

Reply:
This could cause a lifetime issue in coroutines, as reference parameters are not copied over.

Reply:
Ahhhh right, thanks for the info! :)

Reply:
Indeed! dpp::job runs in parallel so reference parameters may become dangling. See the red box in https://deploy-preview-763--dpp-dev.netlify.app/structdpp_1_1job.html

Reply:
Thank you!! Learn something new everyday hahaha